resources/js/components/view/DataImporter.vue

```diff
@@ -0,0 +1,668 @@
+                    :update="state.validationData.update"
+                    :create="state.validationData.create"
+                />
+                <footer class="d-flex align-items-center justify-content-between">
+                    <button
+                        class="btn btn-danger"
+                        @click="cancelImport"
+                    >
+                        {{ t(`global.cancel`) }}
+                    </button>
+
+                    <LoadingButton
+                        v-if="!state.validated"
+                        :loading="state.validating"
+                        class="btn btn-primary"
+                        :disabled="!canValidate || state.validating"
+                        @click="validate"
+                    >
+                        {{ t(`main.importer.validate`) }}
+                    </LoadingButton>
+                    <LoadingButton
+                        v-else
+                        :loading="state.uploading"
+                        class="btn btn-primary"
+                        :disabled="!canImport || state.uploading"
+                        @click="upload"
+                    >
+                        {{ t(`main.importer.import_btn`) }}
+                    </LoadingButton>
+                </footer>
+            </div>
+        </div>
+    </div>
+</template>
+
+
+<script>
+    import {
+        computed,
+        reactive,
+        watch,
+    } from 'vue';
+
+    import store from '%store';
+
+    import { stringSimilarity } from 'string-similarity-js';
+
+    import VueUploadComponent from 'vue-upload-component';
+    import CsvTable from '../CsvTable.vue';
+    import ImporterUpdateState from '../tools/importer/ImporterUpdateState.vue';
+    import EntityImporterSettings from '@/components/tools/importer/EntityImporterSettings.vue';
+    import EntityAttributeMapping from '../tools/importer/EntityAttributeMapping.vue';
+
+    import { useI18n } from 'vue-i18n';
+    import { useToast } from '@/plugins/toast.js';
+
+    import {
+        importEntityData,
+        validateEntityData,
+    } from '@/api.js';
+
+    import {
+        showImportError,
+    } from '@/helpers/modal.js';
+
+    import {
+        translateConcept,
+    } from '@/helpers/helpers.js';
+
+    import {
+        useOptionalLocalStorage
+    } from '../../composables/local-storage';
+    import { onMounted } from 'vue';
+    import LoadingButton from '../forms/button/LoadingButton.vue';
+    import { nextTick } from 'process';
+
+    export default {
+        components: {
+            CsvTable,
+            FileUpload: VueUploadComponent,
+            EntityImporterSettings,
+            EntityAttributeMapping,
+            LoadingButton,
+            ImporterUpdateState,
+        },
+        props: {
+            debug: {
+                type: Boolean,
+                required: false,
+                default: false
+            },
+        },
```

Comment:
remove debug prop?

Reply:
I wouldn't as it is a great tool to populate the fields without having to do all the clicks to get back to the previous state.

Reply:
But this is just for the sake of debugging. I'd not add any non-production logic that might be hard to read/understand and is also a possible error source. We also have to take care of it in case of a framework update. And debugging a specific component is a really rare case.

Reply:
Removed in 2cc4f12b53ce12ee3e8e88324e76ad265bc333f8